app/services/course_option_check.rb

```diff
@@ -0,0 +1,35 @@
+class CourseOptionCheck
```

Comment:
I think `CourseOptionCheck` is a little too general and implies this is general-purpose course option validation to be re-used, when, in fact, it is very specific to the new wizard flow.

Everything else in the app expects just a `course_option`, not a combination of attributes, so I'd rename this to `OfferedCourseDetailsValidator` or similar. It should probably also be a validator rather than a service, with support for raising exceptions rather than returning errors if required.

Reply:
Agreed. We could do with a better name!